src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,33 +1,65 @@
-export const TodoFilter = () => (
+import React from 'react';
+import { FilterType } from '../../services/variables';
+
+type Props = {
+  filterBy: string,
+  setFilterBy: (option: FilterType) => void,
+  query: string,
+  onSetQuery: (option: string) => void,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  filterBy,
+  setFilterBy,
+  query,
+  onSetQuery,
+}) => (
   <form className="field has-addons">
     <p className="control">
       <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
+        <select
+          value={filterBy}
+          data-cy="statusSelect"
+          onChange={(event: React.ChangeEvent<HTMLSelectElement>) => {
+            setFilterBy(event.target.value as FilterType);
+          }}
+        >
+          {(Object.keys(FilterType) as Array<keyof typeof FilterType>)
+            .map(option => (
+              <option
+                key={option}
+                value={option.toLowerCase()}
+              >
+                {option}
+              </option>
+            ))}
```

Comment:
If the object is called `FilterType`, it is better to name the argument closer to this:
```suggestion
          {(Object.keys(FilterType) as Array<keyof typeof FilterType>)
            .map(type => (
              <option
                key={type}
                value={type.toLowerCase()}
              >
                {type}
              </option>
            ))}
```